dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java

```diff
@@ -463,7 +463,8 @@ private void applySharingDisplayNames( FieldFilterParams<?> params, List<FieldPa
                     ((BaseIdentifiableObject) root).getSharing().getUsers().values()
                         .forEach( ua -> ua.setDisplayName( userService.getDisplayName( ua.getId() ) ) );
                 }
-                else if ( o instanceof BaseIdentifiableObject )
```

Comment:
This `else` clause make the code block skip all sharing objects of nested objects.